CMakeLists.txt

```diff
@@ -14,7 +14,9 @@ include(GNUInstallDirs)
 list(APPEND CMAKE_MODULE_PATH ${CMAKE_CURRENT_SOURCE_DIR}/cmake)
 message("CMAKE_MODULE_PATH=${CMAKE_MODULE_PATH}")
 option(CPM_USE_LOCAL_PACKAGES "Try `find_package` before downloading dependencies" ON)
-include(cmake/CPM.cmake)
+if (NOT ${CMAKE_VERSION} VERSION_LESS "3.14.0")
+    include(cmake/CPM.cmake)
```

Comment:
What happens with the AddCPMPackage commands when cmake_version < 3.14?
I can't see the results in continuous integration because their cmake_version is > 3.14. 
One of the problems with supporting older compilers, etc, is maintaining CI coherent with these promises.

Reply:
Yeah that is functionality that would be lost for cmake users < 3.14. We don't have to guarantee that matplotplusplus will work on < 3.14, but for users that do try, they would be pleasantly surprised.

Reply:
That's fine. What I'm worried is that it might lose more than a functionality. 

If the cmake version < 3.14, It might just crash and not work at all when the script first calls AddCPMPackage. I don't know for sure because I haven't tested it yet (and that's kind of part of the problem) but the build script and the formulas for all package managers depend a lot on AddCPMPackage because it installs all the dependencies the user doesn't have.

If it's not going to work at all anyway, it's better to have an error message of the kind "please update cmake" than "the function AddCPMPackage does not exist". Cmake is usually so simple to update that we can even create an error message that tells the user what to type to update cmake on their system.

Reply:
> Cmake is usually so simple to update that we can even create an error message that tells the user what to type to update cmake on their system.

This may not always be the case for users that do not have admin privileges on their work machines (like me...)

But yeah I agree with you that, if we can't test it, we shouldn't merge it. Especially for build related commits.